### ISSUE TITLE
Fans going to zero

### ISSUE THREAD

**Issue by DaveSmth** [trexminer/T-Rex#558]
Card:  ASUS 3090 TUF Gaming.
t-rex.exe -a ethash -o stratum+ssl://eth-au.flexpool.io:5555 -u blah -p x -w blah --log-path logs\trex.txt --api-bind-http 0.0.0.0:4067 --api-read-only --kernel 4 --dag-build-mode 2 --coin eth --intensity 25 --fan tm:90 --mclock 1275 --lock-cclock 1290

Issue:
Occasionally, the fans are dropping to zero and because they take time to spin back up it leads to memory temps higher than desired.  E.g.  Temps are reaching 100 like in the example below.

...
GPU #0: ASUS NVIDIA RTX 3090 - 113.63 MH/s, [T:52/86C, P:281W, F:75%, E:373kH/W], 1844/1844 R:0%
GPU #0: ASUS NVIDIA RTX 3090 - 102.80 MH/s, [T:53/84C, P:280W, F:53%, E:354kH/W], 1845/1845 R:0%
GPU #0: ASUS NVIDIA RTX 3090 - 90.05 MH/s, [T:58/84C, P:285W, F:0%, E:320kH/W], 1845/1845 R:0%
GPU #0: ASUS NVIDIA RTX 3090 - 89.98 MH/s, [T:63/86C, P:289W, F:0%, E:316kH/W], 1846/1846 R:0%
GPU #0: ASUS NVIDIA RTX 3090 - 90.31 MH/s, [T:68/90C, P:293W, F:42%, E:312kH/W], 1846/1846 R:0%
GPU #0: ASUS NVIDIA RTX 3090 - 93.33 MH/s, [T:72/96C, P:340W, F:0%, E:315kH/W], 1848/1848 R:0%
GPU #0: ASUS NVIDIA RTX 3090 - 109.90 MH/s, [T:69/100C, P:341W, F:71%, E:342kH/W], 1849/1849 R:0%
GPU #0: ASUS NVIDIA RTX 3090 - 122.57 MH/s, [T:60/98C, P:337W, F:92%, E:362kH/W], 1849/1849 R:0%
GPU #0: ASUS NVIDIA RTX 3090 - 123.63 MH/s, [T:55/96C, P:333W, F:95%, E:369kH/W], 1849/1849 R:0%
...drops down to target temp.

Other info:
The problem seems to occur when I'm performing a different activity on the machine, like starting a game up that places more load on the GPU and potentially reduces the load on the card's memory.

Possible solution:
Have a minimum fan speed parameter.  Going below a fan speed of ~60% is typically unecessary (for me at least) when the miner is running and would avoid a fan ramp-up lag time and will be more balanced.

**Comment by trexminer** [comment]
At the moment the minimum fan speed for the auto-fan is configured to 5%. It could be that when the miner sets the fan to 5% the GPU accepts it as signal to stop fan spinning completely. You can check that by setting `--fan 5` and checking what the fan speed ends up being. Could you check that for me please?

**Comment by DaveSmth** [comment]
My appologies, the card is an ASUS ROG Strix 3090 (non-OC model).

I just tested --fan 5 and it definitely turns the fans completely off.

Testing higher fan values:

Up to --fan 30 the fans are off.
At --fan 35, the fans independently each start pulsing on and off each second  E.g.  hwinfo shows fan 1 spins to say ~1000rpm while fan 2 is 0.  Then fan 2 spins to ~1000rpm while fan 1 is 0.  Sometimes both are on for a second, then both off.  Likewise at --fan 40.

At --fan 45 both the fans in hwinfo are stable and are running at about 650rpm and t-rex reports 45%.

**Comment by trexminer** [comment]
Ok, thanks. This behaviour appears GPU model specific and the easiest solution would be, like you suggested, to allow the user to set min/max fan speeds in auto-fan mode. We'll put it onto our TODO list, thanks.

**Comment by lordlightman** [comment]
I experienced a similar problem with my ASUS TUF 3070 OC, so setting min fan speed manually would be great!

**Comment by trexminer** [comment]
@DaveSmth @lordlightman When you have a moment, could you please try this build and see if you're able to workaround your issues by setting min/max fan speeds in auto-fan mode?
Download link: https://trex-miner.com/download/test/t-rex-0.21.5-autofan-minmaxspeed-win.zip
Syntax: `--fan tm:90[60-100]` - this is for Dave's use case, maintain 90 degrees mem temp limiting the fan speeds to 60% - 100% range. See the readme for more details, and ask here if it's not clear.
Thanks!

**Comment by DaveSmth** [comment]
I've just tested it and it works well.  I tested running tasks that would typically cause it to drop to 0, which move the load from the gpu memory to the gpu core that causes the drop.  That drops the memory temp so it slows the fans down and happy to say that it held at the min temp.

I also tried with --fan tm:70[60-100] to check the max fan side of things.  E.g. dropping the target temp from 90 to 70 so the fans work harder and fans did hold at 100%.

So all good in my opinion.  Thanks for the quick change, much appreciated!
